### ISSUE TITLE
Some machine names reverted to original

### ISSUE THREAD

**Issue by sontjer** [tailscale/tailscale#4661]
### What is the issue?

No matter on [management page](https://login.tailscale.com/admin/machines) or in the console by typing `tailscale status`.
[Please note that instance-xxx is the original name](https://ezimg.mirai85.cf/i/2022/05/12/nukfgp-1.webp)

### Steps to reproduce

I don't know whether it affects only mine or others as well.

### Are there any recent changes that introduced the issue?

Not easy to figure out which is which...

### OS

Linux

### OS version

ubuntu 20.04

**Comment by DentonGentry** [comment]
I think we've identified why this is happening and stopped further occurrences. We'll post further updates with findings.

**Comment by sontjer** [comment]
Thanks for the response! Everything seems normal so far!

**Comment by DentonGentry** [comment]
I think this happened over an approximately three day period in the first half of May, and was related to organic growth in the userbase finally pushing the coordination server past a tipping point. One of the impacts was it decided that some hostname records were no longer referenced by anything, and dropped them.

We fixed the specific problem in May, and spent about a month working on revamping the coordination server. It was originally written to be essentially node based:
1. a node connects and asks for a netmap, the coordination server assembles the netmap.
1. the next node to connect, even if it is in the same tailnet, even if it is the same *node*, again assembles a netmap response and sends it.

The coordination server now is tailnet based. It computes the responses which should be sent to nodes on a given tailnet, and then specializes the tailnet-based response for each individual node as it asks. We've gained enough headroom to allow for substantial additional growth in the service before needing to do more.

Marking as fixed.
